Streetcode/Streetcode.BLL/MediatR/Authentication/Register/RegisterHandler.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Text.RegularExpressions;
+using AutoMapper;
+using FluentResults;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Streetcode.BLL.DTO.Authentication.Register;
+using Streetcode.BLL.DTO.Users;
+using Streetcode.BLL.Interfaces.Logging;
+using Streetcode.DAL.Entities.Users;
+using Streetcode.DAL.Enums;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+
+namespace Streetcode.BLL.MediatR.Authentication.Register
+{
+    public class RegisterHandler : IRequestHandler<RegisterQuery, Result<RegisterResponseDTO>>
+    {
+        private readonly ILoggerService _logger;
+        private readonly IMapper _mapper;
+        private readonly IRepositoryWrapper _repositoryWrapper;
+        private readonly UserManager<User> _userManager;
+
+        public RegisterHandler(IRepositoryWrapper repositoryWrapper, ILoggerService logger, IMapper mapper, UserManager<User> userManager)
+        {
+            _repositoryWrapper = repositoryWrapper;
+            _logger = logger;
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+
+        public async Task<Result<RegisterResponseDTO>> Handle(RegisterQuery request, CancellationToken cancellationToken)
+        {
+            User user = _mapper.Map<User>(request.registerRequestDTO);
+            string password = request.registerRequestDTO.Password;
+
+            var uniqueResult = await IsUserUnique(user);
+            if (uniqueResult.IsFailed)
+            {
+                return Result.Fail(uniqueResult.Errors);
+            }
+
+            try

```

Comment:
Separate this code block from line 42 to 61 in private function, as you did with isUserUnique. You should separate code vertically to smaller functions to make different levels of abstractions. Handle ideally should contain minimum code and call smaller private functions that would have all the details. 

You can read more about it in a book "clean code" from uncle bob or I also found this article useful as well `https://mosquitolwz.medium.com/clean-code-quotes-5-formatting-b7bdd4a828d6`